magefile.go

```diff
@@ -1276,6 +1276,12 @@ func collectPackageDependencies(platforms []string, packageVersion string, packa
 			}
 		} else {
 			packedBeats := []string{"agentbeat"}
+			// restrict the dependency list only to agentbeat in this case
+			dependencies = packaging.FilterComponents(dependencies, packaging.WithBinaryName("agentbeat"))
```

Comment:
qq: does the `packaging.WithBinaryName("agentbeat")` need to be `"agentbeat.exe"` for windows builds?

Reply:
The binaryName we are comparing here is the one defined in package.yaml and that is defined without the extension